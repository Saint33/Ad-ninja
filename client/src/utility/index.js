import moment from 'moment';

export const formatDate = (date) => {
    let formattedDate;
    if (moment(date).isSame(moment(), "day")){
        formattedDate = `Сегодня ${moment(date).format("HH:mm")}`
    } else {
        formattedDate = moment(date).format("D MMMM HH:mm")
    }
    return formattedDate;
}

export const memberSince = (date) => {
    return moment(date).format("MMMM YYYY")
}