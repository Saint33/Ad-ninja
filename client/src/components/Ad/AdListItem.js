import React from 'react';
import { Link } from 'react-router-dom';
import { formatDate } from '../../utility';

const AdListItem = ({_id, image, title, price, createdAt}) => (
    <div className="adv-list-item">
        <Link to={`/ad/${_id}`}>
            <img 
                className="adv-list-item__image"
                alt="Обьявление" 
                src={`/api/file/${image}`}
            />
        </Link>
        <div className="adv-list-item__description">
            <Link to={`/ad/${_id}`} className="adv-list-item__description_title">{title}</Link>
            <span className="adv-list-item__description_price">{price} р.</span>
            <span className="adv-list-item__description_date">{formatDate(createdAt)}</span>
        </div>
    </div>

);

export default AdListItem;