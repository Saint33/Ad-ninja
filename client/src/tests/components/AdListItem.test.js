import React from 'react';
import AdListItem from '../../components/Ad/AdListItem';
import { shallow } from 'enzyme';
import { formatDate } from '../../utility'

test('AdListItem renders correctly', () => {

    const fakeAd = {
        _id: '322', 
        image: '41a01da6f831c2b85302cba1a331cfa7', 
        title: 'fake title', 
        price: 300, 
        createdAt: "2018-04-30T12:08:39.410Z"
    }

    const wrapper = shallow(
        <AdListItem {...fakeAd} />
    )

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').prop('src')).toEqual(`/api/file/${fakeAd.image}`)
    expect(wrapper.find('img').prop('src')).toEqual(`/api/file/${fakeAd.image}`)
    expect(wrapper.find('Link').at(0).prop('to')).toEqual(`/ad/${fakeAd._id}`)
    expect(wrapper.find('Link').at(1).prop('to')).toEqual(`/ad/${fakeAd._id}`)
    expect(wrapper.find('.adv-list-item__description_price').text()).toEqual(`${fakeAd.price} Ñ€.`)
    expect(wrapper.find('.adv-list-item__description_date').text()).toEqual(`${formatDate(fakeAd.createdAt)}`)

})