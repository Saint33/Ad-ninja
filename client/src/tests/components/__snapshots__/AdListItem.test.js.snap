// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdListItem renders correctly 1`] = `
<div
  className="adv-list-item"
>
  <Link
    replace={false}
    to="/ad/322"
  >
    <img
      alt="Обьявление"
      className="adv-list-item__image"
      src="/api/file/41a01da6f831c2b85302cba1a331cfa7"
    />
  </Link>
  <div
    className="adv-list-item__description"
  >
    <Link
      className="adv-list-item__description_title"
      replace={false}
      to="/ad/322"
    >
      fake title
    </Link>
    <span
      className="adv-list-item__description_price"
    >
      300
       р.
    </span>
    <span
      className="adv-list-item__description_date"
    >
      30 April 15:08
    </span>
  </div>
</div>
`;
